swan-lake/development-tutorials/ballerina-persist/persist-cli-tool.md

```diff
@@ -156,6 +156,52 @@ Behaviour of the `generate` command,
 - The model definition file should contain at least one entity
 - If the user invokes the command twice, it will not fail. It will generate the files once again.
 
+## Generate the data model by introspecting an existing database [Experimental]
+
+>**Info:** The support for introspection is currently an experimental feature, and its behavior may be subject to change in future releases. Also, the support for introspection is currently limited to the MySQL data store.
+
+The command below generates the data model for an existing database. The generated data model can be used to generate the client API for the existing database after verifying it.
+
+```
+$ bal persist pull --datastore mysql --host localhost --port 3306 --user root --database db
+```
+
+| Command Parameter |                                   Description                                    | Mandatory | Default Value |
+|:-----------------:|:--------------------------------------------------------------------------------:|:---------:|:-------------:|
+|    --datastore    | used to indicate the preferred data store. Currently, only 'mysql' is supported. |    No     |     mysql     |
+|      --host       |                        used to indicate the database host                        |    Yes    |     None      |
+|      --port       |                        used to indicate the database port                        |    No     |     3306      |
```

Comment:
shall we mandate the port here? as other databases have different port numbers

Reply:
As per the physical discussion, it was decided to keep the port non-mandatory since we can set the relevant default port for the particular data store if the port is not set, while we check the data store when loading the drivers. Closing this as resolved.